DNS lookup issues... again

#61 identified DNS resolution issues. I thought that I had resolved the problem but it has cropped up again in a later deployment.

The issue seems in part to relate to the issues in [this StackOverflow post](https://stackoverflow.com/questions/65744565/kubernetes-dns-lookg-not-working-from-worker-node-connection-timed-out-no-ser). I'm using the OS they are having problems with and manually disabling firewalld allows DNS traffic to resolve again. When firewalld is running I get these errors in the core-dns logs:

```text
[kubernetes@bri-master-1 ~]$ k logs  -n kube-system  -l k8s-app=kube-dns
.:53
[INFO] plugin/reload: Running configuration SHA512 = 591cf328cccc12bc490481273e738df59329c62c0b729d94e8b61db9961c2fa5f046dd37f1cf888b953814040d180f52594972691cd6ff41be96639138a43908
CoreDNS-1.9.3
linux/amd64, go1.18.2, 45b0a11
[ERROR] plugin/errors: 2 1078600043239553541.3476086839023400652. HINFO: read udp 10.244.3.2:38188->192.168.1.1:53: read: no route to host
[ERROR] plugin/errors: 2 1078600043239553541.3476086839023400652. HINFO: read udp 10.244.3.2:46630->192.168.1.1:53: read: no route to host
[ERROR] plugin/errors: 2 1078600043239553541.3476086839023400652. HINFO: read udp 10.244.3.2:48617->192.168.1.1:53: read: no route to host
.:53
[INFO] plugin/reload: Running configuration SHA512 = 591cf328cccc12bc490481273e738df59329c62c0b729d94e8b61db9961c2fa5f046dd37f1cf888b953814040d180f52594972691cd6ff41be96639138a43908
CoreDNS-1.9.3
linux/amd64, go1.18.2, 45b0a11
[ERROR] plugin/errors: 2 5029101536271341981.505825649077986866. HINFO: read udp 10.244.1.2:35013->192.168.1.1:53: read: no route to host
[ERROR] plugin/errors: 2 5029101536271341981.505825649077986866. HINFO: read udp 10.244.1.2:47530->192.168.1.1:53: read: no route to host
[ERROR] plugin/errors: 2 5029101536271341981.505825649077986866. HINFO: read udp 10.244.1.2:55532->192.168.1.1:53: read: no route to host
```

It looks like firewalld is blocking the route to the gateway for DNS lookups. I would prefer to have firewalld running but DNS traffic allowed through it. Lets see what we can do.

--- a/Ansible/roles/kubernetes/tasks/configure_worker.yml
+++ b/Ansible/roles/kubernetes/tasks/configure_worker.yml
@@ -1,18 +1,4 @@
 ---
-- name: Ensure the workers can communicate through the firewall
-  firewalld:
-    port: "{{ item }}"
-    permanent: true
-    state: enabled
-  loop: "{{ worker_ports }}"
-  register: firewalld_worker_node
-
-- name: reload_firewall
-  when: firewalld_worker_node.changed
-  systemd:
-    name: firewalld
-    state: reloaded
-
 - name: Join node to Kubernetes master
   command:
     cmd: "{{ kubernetes_join_command }}"

--- a/Ansible/roles/kubernetes/tasks/initialise_master.yml
+++ b/Ansible/roles/kubernetes/tasks/initialise_master.yml
@@ -20,20 +20,6 @@
     owner: "kubernetes"
     group: "kubernetes"
 
-- name: Ensure the master can communicate through the firewall
-  firewalld:
-    port: "{{ item }}"
-    permanent: true
-    state: enabled
-  loop: "{{ master_ports }}"
-  register: firewalld_master_node
-
-- name: reload_firewall
-  when: firewalld_master_node.changed
-  systemd:
-    name: firewalld
-    state: reloaded
-
 - name: "Install kubernetes pip package"
   become: true
   become_user: kubernetes

--- a/Ansible/roles/kubernetes/tasks/main.yml
+++ b/Ansible/roles/kubernetes/tasks/main.yml
@@ -2,6 +2,12 @@
 - name: "Disable SWAP"
   include_tasks: disable-swap.yml
 
+- name: "Disable firewalld"
+  service:
+    name: firewalld
+    state: stopped
+    enabled: false
+
 - name: "Install dependencies"
   dnf:
     name: [ iproute-tc, nfs-utils ]

